services/tns_submission_queue/tns_submission_queue.py

```diff
@@ -52,6 +52,618 @@ class TNSReportError(Exception):
+        if len(instrument_ids) > 0:
+            all_tns_instruments = [
+                instrument
+                for instrument in all_tns_instruments
+                if instrument.id in instrument_ids
+            ]
+            if len(all_tns_instruments) == 0:
+                raise TNSReportError(
+                    f'No instruments specified for TNSRobot {tnsrobot_id} are accessible to group {tnsrobot_group.group_id}.'
+                )
+
+        # if the submission has a list of instrument_ids to use, we only use those
+        if (
+            submission_request.instrument_ids is not None
+            and len(submission_request.instrument_ids) > 0
+        ):
+            all_tns_instruments = [
+                instrument
+                for instrument in all_tns_instruments
+                if instrument.id in submission_request.instrument_ids
+            ]
+            if len(all_tns_instruments) == 0:
+                raise TNSReportError(
+                    f'No instruments specified for submission with TNSRobot {tnsrobot_id} are accessible to group {tnsrobot_group.group_id}.'
+                )
+
+        # keep the intersection between the streams that this tns robot has access to (if specified) and the streams that this group has access to
+        all_streams = tnsrobot.streams
+
+        if len(stream_ids) > 0:
+            all_streams = [stream for stream in all_streams if stream.id in stream_ids]
+            if len(all_streams) == 0:
+                raise TNSReportError(
+                    f'No streams specified for TNSRobot {tnsrobot_id} are accessible to group {tnsrobot_group.group_id}.'
+                )
+
+        # if the submission has a list of stream_ids to use, we only use those
+        if (
+            submission_request.stream_ids is not None
+            and len(submission_request.stream_ids) > 0
+        ):
+            all_streams = [
+                stream
+                for stream in all_streams
+                if stream.id in submission_request.stream_ids
+            ]
+            if len(all_streams) == 0:
+                raise TNSReportError(
+                    f'No streams specified for submission with TNSRobot {tnsrobot_id} are accessible to group {tnsrobot_group.group_id}.'
+                )
+
+        # FETCH THE PHOTOMETRY
+        instrument_ids = [instrument.id for instrument in all_tns_instruments]
+        photometry = session.scalars(
+            Photometry.select(user).where(
+                Photometry.obj_id == obj_id,
+                Photometry.instrument_id.in_(instrument_ids),
+            )
+        ).all()
+
+        if len(photometry) == 0:
+            raise TNSReportError(
+                f'No photometry available for {obj_id} with instruments {instrument_ids}.'
+            )
+
+        # FILTER THE PHOTOMETRY BY STREAMS
+        # we keep the photometry that is attached to no streams or to at least one of the streams that are in the list of all_streams
```

Comment:
Out of curiosity, are you not concerned about data permissions when submitting to TNS?